app/com/linkedin/drelephant/exceptions/AzkabanClient.java

```diff
@@ -0,0 +1,207 @@
+            httpClient.getConnectionManager().getSchemeRegistry().register(scheme);
+            HttpResponse response = httpClient.execute(httpPost);
+
+            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                throw new RuntimeException("Login attempt failed.\nStatus line: " + response.getStatusLine().toString() + "\nStatus code: " + response.getStatusLine().getStatusCode());
+            }
+            logger.info(parseResponse(response.toString()));
+            String result = parseContent(response.getEntity().getContent());
+            try {
+                jsonObj = new JSONObject(result);
+                if (jsonObj.has("error")) {
+                    throw new RuntimeException(jsonObj.get("error").toString());
+                }
+
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        } catch (ClientProtocolException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        } catch (UnrecoverableKeyException e) {
+            e.printStackTrace();
+        } catch (KeyManagementException e) {
+            e.printStackTrace();
+        } catch (KeyStoreException e) {
+            e.printStackTrace();
+        } finally {
+            httpClient.getConnectionManager().shutdown();
+        }
+        return jsonObj;
+    }
+
+    public void azkabanLogin(String userName, String password) {
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("action", "login"));
+        urlParameters.add(new BasicNameValuePair("username", userName));
+        urlParameters.add(new BasicNameValuePair("password", password));
+
+        try {
+            JSONObject jsonObject = fetchJson(urlParameters);
+
+            if (!jsonObject.has("session.id")) {
+                throw new RuntimeException("Login attempt failed. The session ID could not be obtained.");
+            }
+
+          this.sessionId = jsonObject.get("session.id").toString();
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    String parseResponse(String response) {
+        String newline = System.getProperty("line.separator");
+        return HtmlUtil.toText(response).replaceAll("azkaban.failure.message=", newline + newline);
+    }
+
+
+    /**
+     * Gets the content from the HTTP response.
+     *
+     * @return The content from the response
+     */
+    String parseContent(InputStream response) throws IOException {
+        BufferedReader reader = null;
+        StringBuilder result = new StringBuilder();
+        try {
+            reader = new BufferedReader(new InputStreamReader(response));
+
+            String line = null;
+            while ((line = reader.readLine()) != null) {
+                logger.info(line);
```

Comment:
Again too much information for info level. Change the log level to debug if you really want it